src/I18n/MessagesFileLoader.php

```diff
@@ -172,12 +159,38 @@ public function translationsFolders(): array
         // If space is not added after slash, the character after it remains lowercased
         $pluginName = Inflector::camelize(str_replace('/', '/ ', $this->_name));
         if (Plugin::isLoaded($pluginName)) {
-            $basePath = App::path('locales', $pluginName)[0];
+            $basePath = Plugin::path($pluginName) . 'resources' . DIRECTORY_SEPARATOR . 'locales' . DIRECTORY_SEPARATOR;
             foreach ($folders as $folder) {
                 $searchPaths[] = $basePath . $folder . DIRECTORY_SEPARATOR;
             }
         }
 
         return $searchPaths;
     }
+
+    /**
+     * @param array<string> $folders Folders
+     * @param string $name File name
+     * @param string $ext File extension
+     * @return string|null File if found
+     */
+    protected function translationFile(array $folders, string $name, string $ext): ?string
+    {
+        $file = null;
+
+        $pos = strpos($name, '/');
```

Comment:
I don't think first checking for slash is necessary, can directly use str_replace(). Same in the I18nExtract command too.